vnext/PropertySheets/JSEngine.props

```diff
@@ -9,12 +9,14 @@
     <UseHermes Condition="'$(UseHermes)' == ''">false</UseHermes>
     <!-- This will be true if (1) the client want to use hermes by setting UseHermes to true OR (2) We are building for UWP where dynamic switching is enabled -->
     <IncludeHermes Condition="'$(IncludeHermes)' == '' And ('$(UseHermes)' == 'true' Or '$(ApplicationType)' == 'Windows Store')">true</IncludeHermes>
-    <HermesVersion Condition="'$(HermesVersion)' == ''">0.8.0-ms.0</HermesVersion>   
+    <HermesVersion Condition="'$(HermesVersion)' == ''">0.8.1-ms.4</HermesVersion>   
     <HermesPackage Condition="'$(HermesPackage)' == '' And Exists('$(PkgReactNative_Hermes_Windows)')">$(PkgReactNative_Hermes_Windows)</HermesPackage>
     <HermesPackage Condition="'$(HermesPackage)' == ''">$(SolutionDir)packages\ReactNative.Hermes.Windows.$(HermesVersion)</HermesPackage>
     <!-- TODO: Can we automatically distinguish between uwp and win32 here? -->
     <HermesArch Condition="'$(HermesArch)' == ''">uwp</HermesArch>
     <EnableHermesInspectorInReleaseFlavor Condition="'$(EnableHermesInspectorInReleaseFlavor)' == ''">false</EnableHermesInspectorInReleaseFlavor>
+    <!-- Use Hermes bytecode bundles provided by metro hermes compiler when available -->
+    <EnableHBCBundles Condition="'$(EnableHBCBundles)' == ''">false</EnableHBCBundles>
```

Comment:
I'm a bit confused by the name of this prop. I added a target a while back to the bundling logic when `UseHermes` is enabled, which is meant to overwrite the existing bundle with HBC extensioned bytecode from the Hermes compiler. https://github.com/microsoft/react-native-windows/blob/17ff6f82640c4fb6d73c7ebb83c036eb200fe88c/vnext/PropertySheets/Bundle.Common.targets#L20  Have you observed this doing anything unintended?

I remember seeing a preprocessor definition that enabled loading adjacent scripts next to each other in the same bundle. Does this prop gate HBC usage or is it specific to multi-script scenarios?

Reply:
Or do we ask Metro to give us HBC in dev when this is enabled?

Reply:
I just read the description of the PR. If this prop affects Metro devtime usage only, could we find a way to express that in the name? Like `EnableDevServerHCBBundles`

Reply:
I didn't think bytecode was used in dev scenarios.  Is this a real scenario?

Reply:
Yeah.. This is an experimental feature in metro (https://github.com/facebook/metro/tree/master/packages/metro-hermes-compiler) to server hermes bytecodes ..
This will enable running optimized bytecodes in dev server scenarios ..

I'll rename it to EnableMetroHCBBundles or EnableDevServerHCBBundles (prefer the former)..